expfmt/encode_test.go

```diff
@@ -63,6 +63,45 @@ func TestNegotiate(t *testing.T) {
 	}
 }
 
+func TestNegotiateOpenMetrics(t *testing.T) {
+	tests := []struct {
+		name              string
+		acceptHeaderValue string
+		expectedFmt       string
+	}{
+		{
+			name:              "OM format, no version",
+			acceptHeaderValue: "application/openmetrics-text",
+			expectedFmt:       string(FmtOpenMetrics_1_0_0),
+		},
+		{
+			name:              "OM format, 0.0.1 version",
+			acceptHeaderValue: "application/openmetrics-text;version=0.0.1",
+			expectedFmt:       string(FmtOpenMetrics_0_0_1),
+		},
+		{
+			name:              "OM format, 1.0.0 version",
+			acceptHeaderValue: "application/openmetrics-text;version=1.0.0",
+			expectedFmt:       string(FmtOpenMetrics_1_0_0),
+		},
+		{
+			name:              "OM format, invalid version",
+			acceptHeaderValue: "application/openmetrics-text;version=0.0.4",
+			expectedFmt:       string(FmtText),
```

Comment:
Why?  Shouldn't it be an error?

Reply:
Historically, the fall-back was always the (classic) text format.

Reply:
Also, this is the current behavior if you request a version of OM that is not supported (e.g.: 1.0.0).